docs/reference/api/platform/Error.md

```diff
@@ -1,14 +1,144 @@
+
+**Although we support Posix error codes for compatibility reasons, we highly encourage all future Mbed OS focused implementations to use Mbed OS error definitions, so errors reported work seamlessly with error reporting and handling implementation in Mbed OS.**
+
+### Error context capture
+
+The error handling system in Mbed OS automatically captures the thread context for each error reported into the system. The context captured includes the following information:
+
+- Error status code – Status code.
+- Error address – Captures return address of set error calls.
+- Error value – A context-specific value the caller sets in `MBED_ERROR()/MBED_WARNING()` calls.
+- Thread ID – ID of the current thread; this information is captured only if the system is built with RTOS support.
+- Thread entry address – Entry function for the current thread; this information is captured only if the system is built with RTOS support.
+- Thread stack size – Size of stack; this information is captured only if the system is built with RTOS support.
+- Thread stack mem - Top of stack; this information is captured only if the system is built with RTOS support.
+- Thread current SP – SP value (PSP or MSP based on what’s active); this information is captured only if the system is built with RTOS support.
+- File name and line number - By default, capturing of file name and line number is disabled. If you need this information to be captured, you need to enable it by using the **MBED_CONF_ERROR_FILENAME_CAPTURE_ENABLED** macro.
+
+STDOUT emits this captured context in the case of fatal errors. In the case of warnings, it is recorded by the system, and you can retrieve it later for system diagnostics, external reporting and debugging purposes.
+
+### Error reporting
+
+Mbed OS provides three ways to report an error using the error handling implementation:
+
+1. Using **error()** function to report a fatal error condition - This is the legacy function used for reporting a fatal error. It has been modified to capture the context information. But, it will not capture the address of the caller, module information and also doesn't support reporting specific error codes. Any implementation using this API to report an error will use the error status **MBED_ERROR_UNKNOWN**. Its highly encouraged that all future Mbed-OS focussed implementations use MBED_ERROR()/MBED_ERROR1() or MBED_WARNING()/MBED_WARNING() macros for reporting errors.**
+1. Using **MBED_ERROR()/MBED_ERROR()** macros to report a fatal error with enhanced capability for capturing module reporting the error.
+1. Using **MBED_WARNING()/MBED_WARNING1()** macros to report a non-fatal error with enhanced capability for capturing module reporting the error.
+
+When you report an error using `MBED_ERROR()` or `MBED_ERROR1()` macros, the error is recorded in the error history (see the **error history** section below) with the context. The error information is printed to STDOUT, and the application is terminated.
```

Comment:
You can replace `error history` with [error history](#error-history) to link to the section header (double check the bracket/parentheses ordering I always mix those up...)

Reply:
Will fix this, Thanks for pointing that out.